src/authn/ccloudProvider.ts

```diff
@@ -297,11 +324,13 @@ export class ConfluentCloudAuthProvider implements vscode.AuthenticationProvider
     });
 
     // if any other part of the extension notices that our current CCloud connection transitions from
-    // VALID_TOKEN to INVALID_TOKEN/NO_TOKEN, we need to remove the session and stop polling
+    // VALID_TOKEN to FAILED/NO_TOKEN, we need to remove the session and stop polling
     const ccloudAuthSessionInvalidatedSub: vscode.Disposable = ccloudAuthSessionInvalidated.event(
       async () => {
         logger.debug("ccloudAuthSessionInvalidated event fired");
-        await this.removeSession(CCLOUD_CONNECTION_ID);
+        // don't delete the actual CCloud connection, just remove it from the authentication provider
```

Comment:
Would love more comment here, as to *why* deleting is bad, but "remove it from the auth provider" is good and proper.

Or, perhaps rather talk positively about what we do need to do any why, and don't talk at all about what we don't need to do.

Reply:
Added a bit more in https://github.com/confluentinc/vscode/pull/566/commits/cd11995dc652911159f4dfb4052dc0241318129a and also https://github.com/confluentinc/vscode/pull/566/commits/1de3a30068e4785e0a8c7a38304c05bb98ba06a3